index.bs

```diff
@@ -1370,77 +896,768 @@ What configuration is the device in after it resets?
+    when invoked, MUST return a new {{Promise}} |promise| and run the following steps
+    <a>in parallel</a>:
+    1. <a>Check if the device is configured</a> with <a>this</a> and |promise|
+        and abort these steps if promise is rejected.
+    1. Let |endpointAddress| be <code>|endpointNumber| | 0x80</code> if
+        |direction| is equal to {{USBDirection/"in"}}, and |endpointNumber|
+        otherwise.
+    1. Let |endpoint| be the result of <a>finding the endpoint</a> with
+        |endpointAddress| and <a>this</a>.
+    1. If |endpoint| is <code>null</code>, <a>reject</a> |promise| with a
+        {{NotFoundError}} and abort these steps.
+    1. Issue a <code>ClearFeature(ENDPOINT_HALT)</code> <a>control transfer</a> to
+        the device to clear the halt condition on |endpoint|.
+    1. On failure <a>reject</a> |promise| with a {{NetworkError}}, otherwise
+        <a>resolve</a> |promise|.
+</div>
+
+
+<div algorithm="USBDevice.transferIn(endpointNumber, length)">
+    The <dfn for=USBDevice method>transferIn(|endpointNumber|, |length|)</dfn> method,
+    when invoked, MUST return a new {{Promise}} |promise| and run the following steps
+    <a>in parallel</a>:
+    1. <a>Check if the device is configured</a> with <a>this</a> and |promise|
+        and abort these steps if promise is rejected.
+    1. Let |endpointAddress| be <code>|endpointNumber| | 0x80</code>
+        (i.e {{USBDirection/"in"}} direction).
+    1. Let |endpoint| be the result of <a>finding the endpoint</a> with
+        |endpointAddress| and <a>this</a>.
+    1. If |endpoint| is <code>null</code>, <a>reject</a> |promise| with a
+        {{NotFoundError}} and abort these steps.
+    1. If |endpoint|.{{USBEndpoint/type}} is not equal to {{USBEndpointType/"bulk"}},
+        <a>reject</a> |promise| with an {{InvalidAccessError}} and abort these steps.
+    1. Let |buffer| be a host buffer with space for |length| bytes.
+    1. As appropriate for |endpoint| enqueue a <a lt="bulk transfer">bulk</a> or
+        <a lt="interrupt transfer">interrupt</a> <a>IN transfer</a> on |endpoint|
+        to receive |length| bytes of data from the device into |buffer|.
+    1. Let |bytesTransferred| be the number of bytes written to |buffer|.
+    1. Let |result| be a new {{USBInTransferResult}}.
+    1. If data was received from <a>this</a> create a new {{ArrayBuffer}} containing
+        the first |bytesTransferred| bytes of |buffer| and set
+        <code>|result|.{{USBInTransferResult/data}}</code> to a new {{DataView}}
+        constructed over it.
+    1. If <a>this</a> responded with more than |length| bytes of data set
+        <code>|result|.{{USBInTransferResult/status}}</code> to {{"babble"}}.
+    1. If the transfer ended because |endpoint| is halted set
+        <code>|result|.{{USBInTransferResult/status}}</code> to {{"stall"}}.
+    1. If <a>this</a> acknowledged the complete transfer set
+        <code>|result|.{{USBInTransferResult/status}}</code> to {{"ok"}}.
```

Comment:
It's clearer to say "the device" instead of "this" in these steps:

```suggestion
    1. If data was received from the device create a new {{ArrayBuffer}} containing
        the first |bytesTransferred| bytes of |buffer| and set
        <code>|result|.{{USBInTransferResult/data}}</code> to a new {{DataView}}
        constructed over it.
    1. If the device responded with more than |length| bytes of data set
        <code>|result|.{{USBInTransferResult/status}}</code> to {{"babble"}}.
    1. If the transfer ended because |endpoint| is halted set
        <code>|result|.{{USBInTransferResult/status}}</code> to {{"stall"}}.
    1. If the device acknowledged the complete transfer set
        <code>|result|.{{USBInTransferResult/status}}</code> to {{"ok"}}.
```

Reply:
Updated in https://github.com/WICG/webusb/pull/212/commits/5aaf696938ed1dc073f8cc8da8c809e9e5019f41. I also updated the similar things in `controlTransferIn()`.